### PR TITLE
DU2 Docs Update

### PR DESCRIPTION
**Pull request by fonty1**
@juhah - I'm going top down, got slowed down with the changes to the client WRT the new `DataUnion` object in the client, creation, etc. in Create and Monitor. I'm new to that change so would most appreciate help there.

Roughly: 

- Intro to Data Unions 🦘
- Create a DU with Core 🦘
- Framework Roles 🦘
- Auth & Identity 🦘
- Create & Monitor 🇫🇮
- Join & Withdraw 🦘 🇫🇮
- UX Best Practices 🦘

And generally:
- Data Union Product -> Data Union
- Refer to streamrClient or client as `streamr`

### CODE REVIEW

**Comment by fonty1** [comment]
@juhah I'll ping you on commit to minimize conflicts.

File: app/src/docs/content/dataUnions/frameworkRoles.mdx [hunk 1]
```diff
@@ -1,34 +1,17 @@
 # Data Unions Framework Roles
 
-The Data Union framework allows app developers to aggregate and monetise the real-time data their app users create whilst sharing revenue with them. The data from many different users are pooled into products offered on the Streamr Marketplace. 
+Each Data Union has one Admin (owner), and `n` data producing members.
```
Comment (juhah) [inline app/src/docs/content/dataUnions/frameworkRoles.mdx:3]: `n` = arbitrary, numerous, large amount of?

File: app/src/docs/content/dataUnions/frameworkRoles.mdx [hunk 2]
```diff
@@ -1,34 +1,17 @@
...
-
-The Administrators are incentivized to perform this work by the `adminFee`, a fraction of incoming revenue. Administrators can set the `adminFee`, and members agree to it by joining the Data Union. Already incurred earnings are of course not affected by changes in the `adminFee`.
-
-Typically, Administrators are initially the creators of the data-producing application and the Data Union Product, though this role can be assigned to someone else or even performed by the members themselves via a DAO.
+Admins have the power to add and remove members. They are responsible for maintaining their Data Unions, including ensuring good data quality and removing members that are not contributing data as they're expected to.
 
-### Operator and Validators
+The Admins are incentivized to perform this work by the `adminFee`, a fraction of the incoming Data Union revenue. Admins can set the `adminFee`, and members agree to it by joining the Data Union. Previous earnings are not affected by changes in the `adminFee`.
```
Comment (juhah) [inline app/src/docs/content/dataUnions/frameworkRoles.mdx:9]:
```suggestion
Admins are incentivized to perform this work by the `adminFee`, a fraction of the incoming Data Union revenue. Admins can set the `adminFee`, and members agree to it by joining the Data Union. Previous earnings are not affected by changes in the `adminFee`.
```

File: app/src/docs/content/dataUnions/frameworkRoles.mdx [hunk 3]
```diff
@@ -1,34 +1,17 @@
...
-
-To withdraw tokens from the `DataunionVault` onto their own on-chain account, a member can query a proof from the Operator (or a Validator) and submit it to the smart contract by doing a normal Ethereum transaction (which requires ETH to pay for the gas). The smart contract verifies the proof and releases the tokens.
+### Members
 
-The Streamr API provides an easy way to query the Operator of each Data Union, although any Validator could be queried instead.
+Members are the people and devices that produce data (via the data-producing application) into the data streams associated with the Data Union. After joining, they start receiving a share of revenue from the Marketplace subscriptions. 
 
-The Administrator is the owner of the smart contract as well as the Marketplace product.
+Members are identified by an Ethereum address, and they join the Data Union by doing a simple HTTP request as described in the Work with Data Unions page.
```
Comment (juhah) [inline app/src/docs/content/dataUnions/frameworkRoles.mdx:17]:
```suggestion
Members are identified by an Ethereum address, and they join the Data Union using the Streamr Client or by doing a simple HTTP request as described in the Join & Withdraw page.
```

Comment (juhah) [reply app/src/docs/content/dataUnions/frameworkRoles.mdx:17]: + link to Join & Withdraw page?

File: app/src/docs/content/dataUnions/introToDataUnions.mdx [hunk 4]
```diff
@@ -14,11 +14,21 @@ import RevenueAndDataMobile2x from './images/intro_to_data_unions_01_revenue_and
 
 # Introduction to Data Unions
 
+The Data Union framework is a data crowdsourcing/crowdselling solution built as part of the Streamr project. Working in tandem with the Streamr Network and Ethereum, the framework powers apps that enable people to earn by sharing valuable data. It enables app developers to aggregate and monetise the real-time data their app users create whilst sharing revenue with them. The Data Union is listed on the Streamr Marketplace as a type of data product for sale. 
+
+This process is called crowdselling. When someone subscribes to the product, the revenues, in DATA token, are automatically distributed among all of the Data Union members — the individual data producers. The framework defines various roles, which we’ll describe next.
```
Comment (juhah) [inline app/src/docs/content/dataUnions/introToDataUnions.mdx:19]:
```suggestion
When someone subscribes to the product, the revenues, in DATA token, are automatically distributed among all of the Data Union members — the individual data producers. The framework defines various roles, which are described below.
```

Comment (juhah) [reply app/src/docs/content/dataUnions/introToDataUnions.mdx:19]: crowdselling already mentioned above

File: app/src/docs/content/dataUnions/introToDataUnions.mdx [hunk 5]
```diff
@@ -14,11 +14,21 @@ import RevenueAndDataMobile2x from './images/intro_to_data_unions_01_revenue_and
...
 
-A Data Union Product is an Ethereum smart contract, created and deployed via the Core app and published on the Marketplace. Most revenue from a Data Union flows to its members, but the admin can set a percentage fee they receive in return for creating and managing the union. For more information on structure, see <Link to={docsLinks.frameworkRoles}>Data Union Framework Roles</Link>.
+A Data Union is an Ethereum smart contract, created and deployed via the Core app and published on the Marketplace. Most revenue from a Data Union flows to its members, but the admin can set a percentage fee they receive in return for creating and managing the union. For more information on structure, see <Link to={docsLinks.frameworkRoles}>Data Union Framework Roles</Link>.
+
+### Smart Contracts
+
+Data Unions is fully on-chain and implemented as smart contracts. While the gas costs on Ethereum mainnet are currently high, Data Unions can run on any Ethereum-compatible sidechain for scalability and affordable cost of operation. The first supported sidechain is <a href="https://www.xdaichain.com" target="_blank" rel="noopener noreferrer">xDai</a>, and other options may be added in the future.
```
Comment (juhah) [inline app/src/docs/content/dataUnions/introToDataUnions.mdx:29]:
```suggestion
Data Unions are fully on-chain and are implemented as smart contracts. Even if the gas costs on Ethereum mainnet are sometimes high, Data Unions can run on any Ethereum-compatible sidechain for scalability and affordable cost of operation. The first supported sidechain is <a href="https://www.xdaichain.com" target="_blank" rel="noopener noreferrer">xDai</a>, and other options may be added in the future.
```

File: app/src/docs/content/dataUnions/introToDataUnions.mdx [hunk 6]
```diff
@@ -14,11 +14,21 @@ import RevenueAndDataMobile2x from './images/intro_to_data_unions_01_revenue_and
...
+
+The benefits of having Data Unions fully on-chain are many. First of all, the security of the system becomes as strong as the security of the underlying blockchain itself. Secondly, Data Union members and builders get to benefit from composability, network effects, and services available on the same blockchain, particularly DeFi platforms.
```
Comment (juhah) [inline app/src/docs/content/dataUnions/introToDataUnions.mdx:31]:
```suggestion
There are numerous benefits of having Data Unions fully on-chain. First of all, the security of the system becomes as strong as the security of the underlying blockchain itself. Secondly, Data Union members and builders get to benefit from composability, network effects, and services available on the same blockchain, particularly DeFi platforms.
```

File: app/src/docs/content/dataUnions/frameworkRoles.mdx (hunk 1 above)
Comment (fonty1) [reply app/src/docs/content/dataUnions/frameworkRoles.mdx]: Yep, added "(many)" to that

**Comment by fonty1** [comment]
@juhah - for me this is ready to merge at your convenience.

Tim added the typescript generated docs to Github Pages FYI, a handy reference (also included in these docs) -  https://streamr-dev.github.io/streamr-client-javascript/classes/dataunion.html

**Pull request merged**